File: Darts/src/burlton/dartzee/code/screen/ai/AbstractAIConfigurationSubPanel.java
package burlton.dartzee.code.screen.ai;

import javax.swing.JPanel;

import burlton.dartzee.code.ai.AbstractDartsModel;
import burlton.dartzee.code.ai.DartsModelNormalDistribution;

/**
 * 
 * @author alexb
 *
 */
public abstract class AbstractAIConfigurationSubPanel extends JPanel
{
	public abstract boolean valid();
	public abstract void populateModel(AbstractDartsModel model);
	public abstract void initialiseFromModel(AbstractDartsModel model);
	
	public void reset()
	{
		initialiseFromModel(new DartsModelNormalDistribution());
	}
}

File: Darts/src/burlton/dartzee/code/screen/ai/AbstractAIConfigurationSubPanel.kt
package burlton.dartzee.code.screen.ai

import burlton.dartzee.code.ai.AbstractDartsModel
import burlton.dartzee.code.ai.DartsModelNormalDistribution
import javax.swing.JPanel

abstract class AbstractAIConfigurationSubPanel : JPanel()
{
    abstract fun valid(): Boolean
    abstract fun populateModel(model: AbstractDartsModel)
    abstract fun initialiseFromModel(model: AbstractDartsModel)

    fun reset()
    {
        initialiseFromModel(DartsModelNormalDistribution())
    }
}
